rdkit/Chem/Draw/__init__.py

```diff
@@ -427,6 +429,257 @@ def ReactionToImage(rxn, subImgSize=(200,200),**kwargs):
+  minx = maxx = miny = maxy = None
+  for i,path in enumerate(path_strings):
+    if path and path[0] == "M":
+      try:
+        for x,y  in [map(float,el.split(",")) for el in path[1:].split(" ") if el]:
+          if minx is None or x < minx:
+            minx = x
+          if miny is None or y < miny:
+            miny = y
+          if maxx is None or x > maxx:
+            maxx = x
+          if maxy is None or y > maxy:
+            maxy = y
+      except ValueError:
+        pass
+
+  # find text areas, assume text is left justified
+  text = [(text.getAttribute('x'), text.getAttribute('y'),
+           text.getAttribute('style'),
+           text.getElementsByTagName('svg:tspan'))
+          for text in doc.getElementsByTagName("svg:text")]
+
+  # assumes fixed width, might be bogus, should at least be greater
+  for x,y,style,tspans in text:
+    try:
+      x = float(x)
+      y = float(y)
+      fontSize = max( list(map(float, fontSizePattern.findall(style))) + [0] )
+      # get the text width
+      count = 0
+      for tspan in tspans:
+        for child in tspan.childNodes:
+          if child.nodeValue:
+            count += len(child.nodeValue)
+            
+      if not count:
+        count = 1
+      xfontSize = fontSize*count
+      if minx is None or x < minx:
+        minx = x
+      if miny is None or y-fontSize < miny:
+        miny = y - fontSize
+      if maxx is None or x+xfontSize > maxx:
+        maxx = x + xfontSize
+      if maxy is None or y > maxy:
+        maxy = y
+
+        
+    except:
+      logging.exception()
+      pass
+    
+  if minx is None: minx = 0
+  if maxx is None: maxx = width
+  if miny is None: miny = 0
+  if maxy is None: maxy = height
+  return minx, maxx, miny, maxy
+
+def makeRect(minx, maxx, miny, maxy):
+  """Make an svg rectangle for debugging purposes"""
+  rect = "<svg:rect style='opacity:0.4;fill:#FF0000;stroke:#000000' width='%s' height='%s' x='%s' y='%s'> </svg:rect>" % (
+    maxx-minx, maxy-miny,
+    minx, miny)
+  return rect
+
+
+def ReactionToSVG(rxn, subImgSize=(200,200), stripSVGNamespace=True,
+                  scaleRelative=True,
+                  debugRender=False,
+                  **kwargs):
+  """
+  """
+  matcher = re.compile(r'^(<.*>\n)(<svg:rect .*</svg\:rect>\n)(.*)</svg\:svg>',re.DOTALL)
+  rect_matcher = re.compile("^(.*width=['])[^']*(.*)")
+  num_reactants = rxn.GetNumReactantTemplates()
+  num_products = rxn.GetNumProductTemplates()
+  # figure out sub image sizes
+  # make a copy so we don't obliterate the original
+  rxn = AllChem.ChemicalReaction(rxn)
+  mols = list(rxn.GetReactants()) + list(rxn.GetProducts())
+
+  # get the relative sizes of the molecules
+  relativeSizes = [1.0] * len(mols)
+  
+  num_mols = len(mols)
+  blocks = [''] * num_mols
+  hdr = ''
+  ftr='</svg:svg>'
+  rect = ''
+
+  _cleanupRXN(rxn)
+  xOffset = 0
+
+  # render at a higher resolution
+  actualSize = subImgSize[:]
+  subImgSize = (600.,600.)
```

Comment:
removed

On Wed, Jul 27, 2016 at 11:21 AM, Greg Landrum <notifications@github.com>
wrote:

> In rdkit/Chem/Draw/__init__.py
> <https://github.com/rdkit/rdkit/pull/993#discussion_r72460448>:
>
> > +
> > +  # get the relative sizes of the molecules
> > +  relativeSizes = [1.0] * len(mols)
> > +
> > +  num_mols = len(mols)
> > +  blocks = [''] * num_mols
> > +  hdr = ''
> > +  ftr='</svg:svg>'
> > +  rect = ''
> > +
> > +  _cleanupRXN(rxn)
> > +  xOffset = 0
> > +
> > +  # render at a higher resolution
> > +  actualSize = subImgSize[:]
> > +  subImgSize = (600.,600.)
>
> this shouldn't be here.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/rdkit/rdkit/pull/993/files/bef16cadaae66d36545aefdd9c8e3ad5df1aa2ce#r72460448>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AJbioDL7BmyBSRSIKg9GjQExQDhJC4wLks5qZ3d2gaJpZM4JVe71>
> .
>